routes/main_routes.js

```diff
@@ -129,7 +132,33 @@ module.exports = function (app, passport) {
             res.redirect('/');
         });
 
-    app.use('/:language/tickets/', ticket_routes);
+    // =====================================
+    // JWT =================================
+    // =====================================
+    app.post("/jwt-login", function (req, res) {
+        if (!req.body.email || !req.body.password || req.body.email.length === 0 || req.body.password.length === 0) {
+            res.status(401).json({message: "email or password are empty"});
+        }
+
+        new User({email: req.body.email}).fetch().then(function (user) {
+            if (user) {
+                if (user.validPassword(req.body.password)) {
+                    // from now on we'll identify the user by the id and the id
+                    // is the only personalized value that goes into our token
+                    var payload = {id: user.id};
+                    var token = jwt.sign(payload, apiTokensConfig.token);
```

Comment:
Sounds legit. As I wrote in Slack, this is the basic (but working and pretty safe) JWT implementation. Anyone who wants to continue - is welcome.
PS - the correct PR to review and merge is https://github.com/Midburn/Spark/pull/309 (this one is obselete).